aws/resource_aws_macie2_invitation.go

```diff
@@ -0,0 +1,171 @@
+package aws
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"strings"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/macie2"
+	"github.com/hashicorp/aws-sdk-go-base/tfawserr"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/terraform-providers/terraform-provider-aws/aws/internal/service/macie2/waiter"
+)
+
+func resourceAwsMacie2Invitation() *schema.Resource {
+	return &schema.Resource{
+		CreateWithoutTimeout: resourceMacie2InvitationCreate,
+		ReadWithoutTimeout:   resourceMacie2InvitationRead,
+		DeleteWithoutTimeout: resourceMacie2InvitationDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+		Schema: map[string]*schema.Schema{
+			"disable_email_notification": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+			},
+			"account_id": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"message": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+			},
+			"invited_at": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func resourceMacie2InvitationCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	conn := meta.(*AWSClient).macie2conn
+
+	accountID := d.Get("account_id").(string)
+	input := &macie2.CreateInvitationsInput{
+		AccountIds: []*string{aws.String(accountID)},
+	}
+
+	if v, ok := d.GetOk("disable_email_notification"); ok {
+		input.DisableEmailNotification = aws.Bool(v.(bool))
+	}
+	if v, ok := d.GetOk("message"); ok {
+		input.Message = aws.String(v.(string))
+	}
+
+	var err error
+	var output *macie2.CreateInvitationsOutput
+	err = resource.RetryContext(ctx, 4*time.Minute, func() *resource.RetryError {
+		output, err = conn.CreateInvitationsWithContext(ctx, input)
+
+		if tfawserr.ErrCodeEquals(err, macie2.ErrorCodeClientError) {
+			return resource.RetryableError(err)
+		}
+
+		if err != nil {
+			return resource.NonRetryableError(err)
+		}
+
+		if len(output.UnprocessedAccounts) != 0 {
+			return resource.NonRetryableError(err)
+		}
```

Comment:
This can actually be removed, and `len(output.UnprocessedAccounts)` will be checked below. `err` will be `nil` here, and that will cause an error in `resource.NonRetryableError(err)`